projects/soft_999_karim_alatrash_fw_102/src/main.c

```diff
@@ -0,0 +1,48 @@
+//
+// Created by Karim Alatrash on 2021-03-08.
+//
+#include <stdbool.h>     // for booleans
+#include <stdint.h>      // for integer types
+#include "interrupt.h"   // interrupts are required for soft timers
+#include "log.h"         // for printing
+#include "soft_timer.h"  // for soft timers
+#include "wait.h"        // for wait function
+
+typedef struct Counter {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counter;
+
+void increment_counter(SoftTimerId timer_id, void *context) {
+  Counter *count = context;
+
+  if (count->counter_a == count->counter_b) {
+    count->counter_a++;
+    LOG_DEBUG("Counter A: %i\n", count->counter_a);
+  } else {
+    count->counter_b++;
+    LOG_DEBUG("Counter B: %i\n", count->counter_b);
+  }
+
+  soft_timer_start_millis(500, increment_counter, count, NULL);
```

Comment:
It's generally a good idea to factor out the 500ms here and below into a `#define` constant. Make sure it has a good descriptive name that ends with the unit as a suffix (in this case `_MS`).